src/Tools/CodeGenerator/Featurizers.yaml

```diff
@@ -1,4 +1,101 @@
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-02-01     | A         | 4     | 1                | 10                |
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-03-01     | A         | 6     | 1                | 7                 |
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-04-01     | A         | 11    | 1                | 5                 |
+            +----------------+-----------+-------+------------------+-------------------+
+
+            A more complex example would be, assuming we have horizon = 2, window size = 2, min window periods = 2, and we want the mean
+            +----------------+-----------+-------+------------------+-------------------+
+            |time(YYYY-MM-DD)| grain     | target| horizon          | target_max        |
+            +================+===========+=======+==================+===================+
+            | 2017-01-01     | A         | 10    | 2                | NAN               |
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-01-01     | A         | 10    | 1                | NAN               |
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-02-01     | A         | 4     | 2                | NAN               |
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-02-01     | A         | 4     | 1                | NAN               |
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-03-01     | A         | 6     | 2                | NAN               |
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-03-01     | A         | 6     | 1                | 7                 |
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-04-01     | A         | 11    | 2                | 7                 |
+            +----------------+-----------+-------+------------------+-------------------+
+            | 2017-04-01     | A         | 11    | 1                | 5                 |
+            +----------------+-----------+-------+------------------+-------------------+
+            Horizon column should be generated by MaxHorizonFeaturizer in the pipeline prior to AnalyticalRollingWindowFeaturizer
+
+    templates:
+      - name: T
+        types:
+            - int8
+            - int16
+            - int32
+            - int64
+            - uint8
+            - uint16
+            - uint32
+            - uint64
+            - float
+            - double
+
+    type_mappings:
+      - input_type: T

```

Comment:
>T [](start = 20, length = 1)

Should this be a tuple as described on line 11?

Reply:
Line 11 I was trying to explain the input to the execute method. This type T is the template type. Would there be a better way for me to make that more clear? Or is this why you were talking about the execute method parameters with me?

Reply:
I think that the input type will end up being `tuple<datetime, vector<string>, horizon, T>`. The output will be `T`, where the output type is flagged as optional.

Reply:
Is horizon going to be a type as well? I have it listed as uint8 now but can update and push again if needed.

Reply:
The horizon value will not be a template, but we will need to to be the same across MaxHorizon, RollingWindow, and LagLead. Let's make it uint16.

Reply:
Per the discussion we had, making this uint32.